File: Python/Euler002.py
#!/usr/bin/env python3
from __future__ import print_function
from itertools import takewhile


def main():
    fibs = fibonacci()
    evens = (fib for fib in fibs if fib % 2 is 0)
    under_4m = takewhile(lambda x: x < 4000000, evens)
    print(sum(under_4m))


def fibonacci():
    previous = 1
    current = 1
    yield previous
    yield current

    while True:
        old_prev = previous
        previous = current
        current = previous + old_prev
        yield current


if __name__ == '__main__':
    main()

File: Node/Euler002.js
#!/usr/bin/env node

function main() {
  let sum = 0;

  let previous = 1;
  let current = 1;
  while (current < 4000000) {
    let oldPrev = previous;
    previous = current;
    current = previous + oldPrev;

    if (current % 2 == 0) {
      sum += current;
    }
  }

  console.log(sum);
}

if (require.main === module) {
  main();
}
